### ISSUE TITLE
Nixpkgs config is ignored

### ISSUE THREAD

**Issue by expipiplus1** [xzfc/cached-nix-shell#13]
To reproduce:

- Allow broken packages in `~/.config/nixpkgs/config.nix
    ```nix
    { nixpkgs.config.allowBroken = true; }
    ```

- Run `cached-nix-shell -p haskellPackages.alga` (or any other package marked as broken)

- Observe that the package doesn't built with the following error

```
$ cached-nix-shell -p haskellPackages.alga
cached-nix-shell: updating cache
error: Package ‘alga-0.2.2’ in /home/j/src/nixpkgs/pkgs/development/haskell-modules/hackage-packages.nix:25062 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
cached-nix-shell: nix-shell: exit code: 1
```

- Run `nix-shell -p haskellPackages.alga` and observe that the package builds correctly

This happens for `allowUnfree` too

**Comment by xzfc** [comment]
Fixed in 422036e95e30b25f8c67e44a45f779044ce0407e.

**Comment by expipiplus1** [comment]
Thanks again!
